### PR TITLE
fix(BusinessPartnerNumber): adjusted regex to match CX-0010

### LINKED ISSUE #570: [Model Update]: Fix wrong bpns regexes in Shared BPN model
## Update Reason

Currently BPN regex is incorrect as it does not allow alphanumerical characters in 10 digit area

Format according to standard CX-0010
image

Format according to regex (example BPNL)
`^BPNL[0-9]{8}[a-zA-Z0-9]{4}$`

Propsed Regex: `^BPNL[a-zA-Z0-9]{12}$`

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists
- [x] The potential updates are discussed with all stakeholders
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [x] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

 -->

Closes #570 

Did not include SAMM update

Major version update needed? Overall it's less restricitve why it could also be a minor version change

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [ ] All required reviewers have approved this PR (see reviewers section)
- [ ] The new aspect (version) will be implemented by at least one data provider
- [ ] The new aspect (version) will be consumed by at least one data consumer
- [ ] There exists valid test data
- [ ] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [ ] The model has at least version '1.0.0'
- [ ] If a previous model exists, model deprecation has been checked for previous model
- [ ] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

File: io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+#
+# This work is made available under the terms of the
+# Creative Commons Attribution 4.0 International (CC-BY-4.0) license,
+# which is available at
+# https://creativecommons.org/licenses/by/4.0/legalcode.
+#
+# SPDX-License-Identifier: CC-BY-4.0
+#######################################################################
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.0.0#>.
```
Comment (agg3fe) [inline io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl:15]: samm:meta-model 2.1.0 required in all new models.

File: io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#######################################################################
+# Copyright (c) 2023 Contributors to the Eclipse Foundation
```
Comment (catroest) [inline io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl:2]: 2023-2024

Comment (tom-rm-meyer-ISST) [reply io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl]: well seen!

File: io.catenax.shared.business_partner_number/RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,6 +1,16 @@
 # Changelog
 All notable changes to this model will be documented in this file.
 
+## [2.0.0] - 2024-01-XY
```
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply io.catenax.shared.business_partner_number/RELEASE_NOTES.md]: updated

File: io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl (hunk 1 above)
Comment (tom-rm-meyer-ISST) [reply io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl]: updated all resepective references stating to namepsace org.eclipse.esmf.samm

File: io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    samm:properties (:bpnlProperty :bpnsProperty :bpnaProperty);
+    samm:operations ();
+    samm:events ().
+:bpnlProperty a samm:Property;
+    samm:preferredName "Business Partner Number Legal Entity Property"@en;
+    samm:description "Property based on a BPNL."@en;
+    samm:characteristic :BpnlTrait;
+    samm:exampleValue "BPNL0123456789ZZ".
+:BpnlTrait a samm-c:Trait;
+    samm:preferredName "BPNl Trait"@en;
```
Comment (catroest) [inline io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl:36]: can you make the "L" in upper letters? it looks like an "I"

**Comment by agg3fe** [comment]
@tom-rm-meyer-ISST are we ready for this model? I will directly merge it. let me know.

File: io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl (hunk 4 above)
Comment (tom-rm-meyer-ISST) [reply io.catenax.shared.business_partner_number/2.0.0/BusinessPartnerNumber.ttl]: Updated

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by tom-rm-meyer-ISST** [comment]
@agg3fe, model is ready from my point of view

**Pull request merged**